src/backend/distributed/operations/worker_shard_copy.c

```diff
@@ -103,10 +119,17 @@ ConnectToRemoteAndStartCopy(ShardCopyDestReceiver *copyDest)
 														 NULL /* database (current) */);
 	ClaimConnectionExclusively(copyDest->connection);
 
+	/* Setup Replication Origin Session if not setup already for
+	 * avoiding publication of events more than once. */
+	char replicationOrginSetupCommand[REPLICATION_ORIGIN_CMD_BUFFER_SIZE];
+	int originId = GetLocalNodeId();
+	snprintf(replicationOrginSetupCommand, REPLICATION_ORIGIN_CMD_BUFFER_SIZE,
+			 CDC_REPLICATION_ORIGIN_SESION_SETUP_CMD, originId);
```

Comment:
we also need to set the replication origin in the local backend when useLocalCopy is true

Reply:
(removing useLocalCopy may also be an option, if we understand the implications)

Reply:
Added the replication origin session management for local sessions seperately for local shard copy.